internal/app/storage/in_memory.go

```diff
@@ -0,0 +1,38 @@
+package storage
+
+import (
+	"errors"
+	"github.com/FeelDat/urlshort/internal/utils"
+	"math/rand"
+)
+
+type Repository interface {
+	ShortenURL(fullLink string) string
+	GetFullURL(shortLink string) (string, error)
+}
+
+type InMemoryStorage struct {
+	Links map[string]string
+}
+
+func NewInMemoryStorage() Repository {
+	return &InMemoryStorage{
+		Links: make(map[string]string),
+	}
+}
+
+func (mapStorage *InMemoryStorage) ShortenURL(fullLink string) string {
```

Comment:
mapStorage  - так как эта переменная для структуры и видна только в этом пакете, можно и стоит выбирвть более короткие названия
```suggestion
func ( s *InMemoryStorage) ShortenURL(fullLink string) string {
```